src/m3ninx/index/regexp_test.go

```diff
@@ -0,0 +1,284 @@
+		testCase{
+			name:           "concat of sub expressions with only legal ^ and $",
+			input:          "(^abc$)(?:^xyz$)",
+			expectedOutput: "(abc(?-m:$))\\Axyz",
+		},
+		testCase{
+			name:           "alternate of sub expressions with illegal ^ and $",
+			input:          "(?:^a$bc$)|(?:^xyz$)",
+			expectedOutput: "a(?-m:$)bc|xyz",
+		},
+		testCase{
+			name:           "concat of sub expressions with illegal ^ and $",
+			input:          "(?:^a$bc$)(?:^xyz$)",
+			expectedOutput: "a(?-m:$)bc(?-m:$)\\Axyz",
+		},
+		testCase{
+			name:           "question mark case both boundaries success",
+			input:          "(?:^abc$)?",
+			expectedOutput: "(?:abc)?",
+		},
+		testCase{
+			name:           "question mark case only ^",
+			input:          "(?:^abc)?",
+			expectedOutput: "(?:abc)?",
+		},
+		testCase{
+			name:           "question mark case only $",
+			input:          "(?:abc$)?",
+			expectedOutput: "(?:abc)?",
+		},
+		testCase{
+			name:           "question concat case $",
+			input:          "abc$?",
+			expectedOutput: "abc",
+		},
+		testCase{
+			name:           "star mark case both boundaries success",
+			input:          "(?:^abc$)*",
+			expectedOutput: "(?:abc)?(?:\\Aabc(?-m:$))*",
+		},
+		testCase{
+			name:           "star mark case only ^",
+			input:          "(?:^abc)*",
+			expectedOutput: "(?:abc)?(?:\\Aabc)*",
+		},
+		testCase{
+			name:           "star mark case only $",
+			input:          "(?:abc$)*",
+			expectedOutput: "(?:abc)?(?:abc(?-m:$))*",
+		},
+		testCase{
+			name:           "star concat case $",
+			input:          "abc$*",
+			expectedOutput: "abc(?-m:$)*",
+		},
+		testCase{
+			name:           "star concat case ^",
+			input:          "^*abc",
+			expectedOutput: "\\A*abc",
+		},
+		testCase{
+			name:           "plus mark case both boundaries success",
+			input:          "(?:^abc$)+",
+			expectedOutput: "abc(?:\\Aabc(?-m:$))*",
+		},
+		testCase{
+			name:           "plus mark case with capturing group",
+			input:          "(^abc$)+",
+			expectedOutput: "(abc)(\\Aabc(?-m:$))*",
+		},
+		testCase{
+			name:           "plus mark case only ^",
+			input:          "(?:^abc)+",
+			expectedOutput: "abc(?:\\Aabc)*",
+		},
+		testCase{
+			name:           "plus mark case only $",
+			input:          "(?:abc$)+",
+			expectedOutput: "abc(?:abc(?-m:$))*",
+		},
+		testCase{
+			name:           "plus concat case $",
+			input:          "abc$+",
+			expectedOutput: "abc(?-m:$)*",
+		},
+		testCase{
+			name:           "plus concat case ^",
+			input:          "^+abc",
+			expectedOutput: "\\A*abc",
+		},
+		testCase{
+			name:           "repeat case both boundaries success",
+			input:          "(?:^abc$){3,4}",
+			expectedOutput: "abc(?:\\Aabc(?-m:$)){2,3}",
+		},
+	}
+	for _, tc := range testCases {
+		re, err := parseRegexp(tc.input)
+		require.NoError(t, err)
+		parsed, err := ensureRegexpUnanchored(re)
+		require.NoError(t, err)
+		assert.Equal(t, tc.expectedOutput, parsed.String(), tc.name)
```

Comment:
nit: Want to make this:
```go
t.Run(tc.name, func(t *testing.T) {
  // test...
})
```

So that we get the test names for each one and show which ones pass/fail individually?